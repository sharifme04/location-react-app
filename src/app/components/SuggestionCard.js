import React from "react";
import PropTypes from "prop-types";

const SuggestionCard = ({
  locations,
  handlebothLocation,
  handleSuggestion,
  display,
}) => {
  return (
    locations?.length && (
      <div className="suggestion-card" style={{ display: display }}>
        <div className="card" style={{ width: "21rem" }}>
          <div className="loading">
            <div className="d-flex align-items-center">
              <div
                className="spinner-border ms-auto"
                role="status"
                aria-hidden="true"
              ></div>
            </div>
            <p>Standort verwenden</p>
          </div>
          <ul className="list-group list-group-flush">
            {locations.length &&
              locations.map((e, i) => (
                <li
                  className="list-group-item"
                  key={e?.id + i}
                  onClick={() => {
                    handlebothLocation(e);
                    handleSuggestion("none");
                  }}
                >
                  <p className="area">{e?.objectName}</p>
                  <p className="city">{e?.locality}</p>
                </li>
              ))}
          </ul>
        </div>
      </div>
    )
  );
};

SuggestionCard.propTypes = {
  handlebothLocation: PropTypes.func,
  handleSuggestion: PropTypes.func,
  display: PropTypes.string,
  locations: PropTypes.array,
};

export default SuggestionCard;
