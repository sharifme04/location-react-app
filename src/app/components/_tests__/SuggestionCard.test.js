import React from "react";
import renderer from "react-test-renderer";
import SuggestionCard from "../SuggestionCard";

const locations = [
  {
    "#text": "Stadt",
    anyType: "poi",
    anyTypeSort: "4",
    buildingName: "",
    buildingNumber: "",
    id: "2002734",
    listIndex: "0",
    locality: "Filderstadt",
    mainLocality: "",
    mapName: "EPSG:4326",
    matchQuality: "1000",
    nameKey: "",
    objectName: "Stadt",
    omc: "8116077",
    placeID: "-1",
    postCode: "",
    selected: "1",
    stateless:
      "poiID:2002734:8116077:-1:Stadt:Filderstadt:Stadt:ANY:POI:3514784:769226:NBWT:VVS",
    streetName: "",
    value: "0:1",
    x: "9.19964",
    y: "48.65440",
  },
];

test("snapshot Form renders with empty props", () => {
  const component = renderer.create(<SuggestionCard />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot Form renders with props", () => {
  const component = renderer.create(
    <SuggestionCard
      handleSuggestion={() => {}}
      handlebothLocation={() => {}}
      locations={[]}
      display={""}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot Form renders with props data", () => {
  const component = renderer.create(
    <SuggestionCard
      handleSuggestion={() => {}}
      handlebothLocation={() => {}}
      locations={[locations]}
      display={"none"}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
