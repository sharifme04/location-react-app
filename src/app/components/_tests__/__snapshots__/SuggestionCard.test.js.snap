// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Form renders with empty props 1`] = `null`;

exports[`snapshot Form renders with props 1`] = `"0"`;

exports[`snapshot Form renders with props data 1`] = `
<div
  className="suggestion-card"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="card"
    style={
      Object {
        "width": "21rem",
      }
    }
  >
    <div
      className="loading"
    >
      <div
        className="d-flex align-items-center"
      >
        <div
          aria-hidden="true"
          className="spinner-border ms-auto"
          role="status"
        />
      </div>
      <p>
        Standort verwenden
      </p>
    </div>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
        onClick={[Function]}
      >
        <p
          className="area"
        />
        <p
          className="city"
        />
      </li>
    </ul>
  </div>
</div>
`;
